### PR TITLE
[LG] re-do multiLingualLg

### PR DESCRIPTION
**Pull request by Danieladu**
close: #3551 
- add a `defaultLanguages` param format list which can replace "" as default
 - sync with latest SDK how language policy is updated, "de-de" => "de-de", "de"
 - replace `EvaluateText` with `Evaluate`
 - support passed in parsed Templates instead of just file paths, `Dict<string, Templates>`

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.LanguageGeneration/MultiLanguageLG.cs [hunk 1]
```diff
@@ -21,10 +21,11 @@ public class MultiLanguageLG
         /// Initializes a new instance of the <see cref="MultiLanguageLG"/> class.
         /// </summary>
         /// <param name="localeLGFiles">A dictionary of locale and LG file.</param>
-        public MultiLanguageLG(Dictionary<string, string> localeLGFiles)
+        /// <param name="defaultLanguages">Default language.</param>
+        public MultiLanguageLG(Dictionary<string, string> localeLGFiles, params string[] defaultLanguages)
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Builder.LanguageGeneration/MultiLanguageLG.cs:25]: lgFilePerLocale or lgFileMap?

Comment (boydc2014) [reply libraries/Microsoft.Bot.Builder.LanguageGeneration/MultiLanguageLG.cs:25]: one default langugage is probabaly good

File: libraries/Microsoft.Bot.Builder.LanguageGeneration/MultiLanguageLG.cs [hunk 2]
```diff
@@ -37,6 +38,22 @@ public MultiLanguageLG(Dictionary<string, string> localeLGFiles)
             }
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MultiLanguageLG"/> class.
+        /// </summary>
+        /// <param name="templatesDict">A dictionary of LG file templates.</param>
+        /// <param name="defaultLanguages">Default language.</param>
+        public MultiLanguageLG(Dictionary<string, Templates> templatesDict, params string[] defaultLanguages)
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Builder.LanguageGeneration/MultiLanguageLG.cs:46]: templastePerLocale

File: libraries/Microsoft.Bot.Builder.LanguageGeneration/MultiLanguageLG.cs [hunk 3]
```diff
@@ -20,23 +18,40 @@ public class MultiLanguageLG
...
+            foreach (var filesPerLocale in filePerLocale)
             {
                 lgPerLocale[filesPerLocale.Key] = Templates.ParseFile(filesPerLocale.Value);
             }
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MultiLanguageLG"/> class.
+        /// </summary>
+        /// <param name="templatesPerLocale">A dictionary of LG file templates.</param>
```
Comment (chrimc62) [inline libraries/Microsoft.Bot.Builder.LanguageGeneration/MultiLanguageLG.cs:42]: >LG [](start = 61, length = 2)

per locale

**Comment by boydc2014** [comment]
Just heads up, this PR addressed a few feedback from VA team, and we've asked @lzc850612 to give a try before merge it.

**Pull request merged**